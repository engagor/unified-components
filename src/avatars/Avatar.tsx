import React, { ReactElement } from 'react';

export type AvatarProps = {
    image: string;
    size: 'small' | 'medium' | 'large';
    alt: string;
    username?: string;
};

export default function Avatar(avatarProps: AvatarProps): ReactElement {
    let width,
        height = 64;

    if (avatarProps.size === 'small') {
        width = height = 16;
    }

    if (avatarProps.size === 'medium') {
        width = height = 32;
    }

    return (
        <div role="img" aria-label={avatarProps.username ?? avatarProps.alt} className="rounded-full overflow-hidden inline-block">
            <img aria-hidden="true" className="block" loading="lazy" src={avatarProps.image} alt={avatarProps.username ?? avatarProps.alt} width={width} height={height} />
        </div>
    );
}
