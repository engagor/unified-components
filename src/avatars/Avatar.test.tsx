import { renderToStaticMarkup } from 'react-dom/server';

jest.mock(
    'react-dom',
    () => {
        return {
            createPortal: (element: ReactElement, node: HTMLElement) => {
                node.innerHTML = renderToStaticMarkup(element);
            },
        }
    }
);

import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

it('renders correctly', async () => {
    const element = document.createElement('div');

    const avatar = await renderer
        .create(<Avatar image="https://via.placeholder.com/150" size="large" alt="alt_text" />)
        .toJSON();

    await expect(avatar).toMatchSnapshot();
    expect(element).toMatchSnapshot();
});
