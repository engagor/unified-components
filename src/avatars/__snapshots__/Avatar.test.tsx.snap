// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-label="alt_text"
  className="rounded-full overflow-hidden inline-block"
  role="img"
>
  <img
    alt="alt_text"
    aria-hidden="true"
    className="block"
    height={64}
    loading="lazy"
    src="https://via.placeholder.com/150"
  />
</div>
`;

exports[`renders correctly 2`] = `<div />`;
